app/lib/task/backend.dart

```diff
@@ -737,8 +738,20 @@ class TaskBackend {
   /// package name, version and randomized blobId.
   Future<BlobIndex?> _taskResultIndex(String package, String version) async =>
       await cache.taskResultIndex(package, version).get(() async {
+        // Don't try to load index if we don't consider the version for analysis.
+        final status = await packageStatus(package);
+        if (!status.versions.containsKey(version)) {
+          return BlobIndex.empty(blobId: '');
+        }
         // Try runtimeVersions in order of age
         for (final rt in acceptedRuntimeVersions) {
+          // skip current runtime if it is a know failure or hasn't been started yet.
+          if (rt == runtimeVersion) {
+            final versionStatus = status.versions[version]!.status;
+            if (versionStatus == PackageVersionStatus.failed) {
+              continue;
+            }
+          }
```

Comment:
I'd suggest moving this outside the loop.

If the current runtimeVersion failed, then failure is intentional. We should only fallback when we haven't processed the current version, because it's pending.